org.eclipse.xtext.xbase.tests/src/org/eclipse/xtext/xbase/tests/validation/XbaseValidationTest.xtend

```diff
@@ -1128,4 +1128,12 @@ class XbaseValidationTest extends AbstractXbaseTestCase {
 			}
 		'''.expression.assertNoErrors
 	}
+
+	@Test def void testInvalidNestedReturnBug406762() {
+		'''
+			{
+				return return 0
```

Comment:
return throw is technically ok and there are situations where you want to allow it.
So I think we shouldn't validate that, or at least only if there is no branch that evaluates to an actual value.

E.g.
return if (true) throw else throw //invalid
return if (baz) throw else 42 // ok